### PR TITLE
Reduce Error-ness of not Finding Solutions

### PR DESCRIPTION
**Pull request by nlordell**
# Description

We recently added the ability for a driver to return multiple solutions. With this, we should no longer return an error when no solutions are found but an empty vector.

This was prompted by the general noisiness in metrics and logs for "regular" behaviour for the solvers.

# Changes

- [x] Return empty solutions list when none is found instead of an error.

## How to test

Driver tests assert an "OK" response with empty solutions instead of errors.

### CODE REVIEW

File: crates/driver/src/infra/api/error.rs [hunk 1]
```diff
@@ -33,7 +33,10 @@ impl From<Kind> for (hyper::StatusCode, axum::Json<Error>) {
         let description = match value {
             Kind::QuotingFailed => "No valid quote found",
             Kind::SolverFailed => "Solver engine returned an invalid response",
-            Kind::SolutionNotFound => "No solution found for the auction",
+            Kind::SolutionNotAvailable => {
+                "no solution is available yet, this might mean that /settle was called before \
+                 /solve returned"
```
Comment (sunce86) [inline crates/driver/src/infra/api/error.rs:38]: How can this happen? (settle being called before solve returned)?

**Review by fleupold — APPROVED** [review]

File: crates/driver/src/infra/observe/mod.rs [hunk 2]
```diff
@@ -182,15 +182,22 @@ pub fn settled(solver: &solver::Name, result: &Result<competition::Settled, comp
...
-        Ok(solved) => {
+        Ok(Some(solved)) => {
             tracing::info!(?solved, "solved auction");
             metrics::get()
                 .solutions
                 .with_label_values(&[solver.as_str(), "Success"])
                 .inc();
         }
+        Ok(None) => {
+            tracing::info!("no solution found");
```
Comment (fleupold) [inline crates/driver/src/infra/observe/mod.rs:195]: not even sure this is worth an info log (maybe debug is fine)

Comment (nlordell) [reply crates/driver/src/infra/observe/mod.rs:195]: Good point.

File: crates/driver/src/infra/api/error.rs (hunk 1 above)
Comment (nlordell) [reply crates/driver/src/infra/api/error.rs:38]: Added a `driver` test to document this.

Comment (nlordell) [reply crates/driver/src/infra/api/error.rs:38]: 9daec2d838cba910f1a737b0e0ff51b146c57188

File: crates/driver/src/tests/setup/mod.rs [hunk 3]
```diff
@@ -850,22 +847,10 @@ impl SolveOk {
     pub fn default_score(self) -> Self {
         self.score_in_range(DEFAULT_SCORE_MIN.into(), DEFAULT_SCORE_MAX.into())
     }
-}
 
-pub struct SolveErr {
```
Comment (nlordell) [inline crates/driver/src/tests/setup/mod.rs:855]: Removed this type as it was no longer used.

**Pull request merged**